src/main/java/AddCommand.java

```diff
@@ -0,0 +1,70 @@
+import java.io.IOException;
+
+public class AddCommand extends Command {
+
+    public AddCommand(String command) {
+        super(command);
+    }
+
+    boolean isExit() {
+        return false;
+    }
+
+    void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        String arr[] = command.split(" ", 2);
+        String firstWord = arr[0];
+
+        if (firstWord.equals("todo")) {
+            if (arr.length > 1) { //check for errors
+
+                //CALL TASKLIST: add task
+                tasks.addTodo(arr[1]);
+                //CALL UI: print output
+                ui.printAddTask(tasks);
+                //CALL STORAGE: write new file
+                storage.addTodo(tasks);
+            } else {
+                throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+            }
+        } else if (firstWord.equals("deadline")) {
+            if (arr.length > 1) { //check for errors
+                //separate task and deadline
+                String arr2[] = arr[1].split("/", 2);
+                if (arr2.length > 1) { //check that deadline is specified
+                    String arr3[] = arr2[1].split(" ", 2);
```

Comment:
You can abstract this line out to achieve SLAP with the line below I believe. It may seems very trivial abstraction, but doing it can have a more readable code and more organized code too.

You can do something like
```String arr3 = getArr3FromArr2(arr2)```, something like this